patroni/dcs/__init__.py

```diff
@@ -447,37 +470,40 @@ def matches(self, name: Union[str, None], check_leader: Optional[bool] = False)
         """
         ret = False
         if name and not self.is_empty:
-            search_str = (self.sync_standby or '') + (',' + self.leader if check_leader else '')
+            search_str = (self.sync_standby or '') + (',' + (self.leader or '') if check_leader else '')
             ret = name.lower() in self._str_to_list(search_str.lower())
         return ret
 
-    def leader_matches(self, name: Union[str, None]) -> bool:
+    def leader_matches(self, name: Optional[str]) -> bool:
         """:returns: `True` if name is matching the `SyncState.leader` value."""
-        return name and not self.is_empty and name.lower() == self.leader.lower()
+        return bool(name and not self.is_empty and name.lower() == (self.leader or '').lower())
+
+
+_HistoryTuple = Union[Tuple[int, int, str], Tuple[int, int, str, str], Tuple[int, int, str, str, str]]
 
 
-class TimelineHistory(namedtuple('TimelineHistory', 'index,value,lines')):
+class TimelineHistory(NamedTuple):
     """Object representing timeline history file"""
+    index: _Version
+    value: Any
+    lines: List[_HistoryTuple]
 
     @staticmethod
-    def from_node(index, value):
+    def from_node(index: _Version, value: str) -> 'TimelineHistory':
```

Comment:
```suggestion
    def from_node(index: _Version, value: Optional[str]) -> 'TimelineHistory':
```

As the `except` blocks handles `TypeError`, I'd say `value` can be `None`.